ext/RMagick/rmimage.c

```diff
@@ -12076,15 +12078,20 @@ resample(int bang, int argc, VALUE *argv, VALUE self)
 
     exception = AcquireExceptionInfo();
 #if defined(IMAGEMAGICK_7)
+    Image *preprocess = NULL;
     if (blur > 1.0)
     {
-        image = BlurImage(image, blur, blur, exception);
+        preprocess = BlurImage(image, blur, blur, exception);
     }
     else if (blur < 1.0)
     {
-        image = SharpenImage(image, blur, blur, exception);
+        preprocess = SharpenImage(image, blur, blur, exception);
+    }
+    new_image = ResampleImage((preprocess ? preprocess : image), x_resolution, y_resolution, filter, exception);
+    if (preprocess != NULL)
```

Comment:
Ah, I guess it makes the check here possible?

Reply:
yep, and `preprocess ? preprocess : image` :)